server/routerlicious/packages/routerlicious-base/src/nexus/runnerFactory.ts

```diff
@@ -56,31 +60,163 @@ export class OrdererManager implements core.IOrdererManager {
 	) {}
 
 	public async getOrderer(tenantId: string, documentId: string): Promise<core.IOrderer> {
-		if (!this.globalDbEnabled) {
-			const messageMetaData = { documentId, tenantId };
-			Lumberjack.info(`Global db is disabled, checking orderer URL`, messageMetaData);
-			const tenant = await this.tenantManager.getTenant(tenantId, documentId);
-
-			Lumberjack.info(
-				`tenant orderer: ${JSON.stringify(tenant.orderer)}`,
-				getLumberBaseProperties(documentId, tenantId),
+		const ordererConnectionType = await this.getOrdererConnectionType(tenantId, documentId);
+		const ordererId = this.getOrdererConnectionMapKey(tenantId, documentId);
+		if (this.ordererConnectionCloseTimeoutMap.has(ordererId)) {
+			// Clear any existing connection timeout because a new connection was added.
+			clearTimeout(this.ordererConnectionCloseTimeoutMap.get(ordererId));
+		}
+
+		if (ordererConnectionType === "local") {
+			Lumberjack.info(`Using local orderer`, getLumberBaseProperties(documentId, tenantId));
+			return this.localOrderManager.get(tenantId, documentId).then((orderer) => {
```

Comment:
Need a catch() for this promise?

Reply:
Updating to use `await` instead of promise chaining to make this code block more clear, per @zhangxin511 's comment. The error will get bubbled up to the caller